wrapanapi/systems/virtualcenter.py

```diff
@@ -225,9 +251,22 @@ def _pick_datastore(self, allowed_datastores):
             key=lambda ds: float(ds.summary.freeSpace) / float(ds.summary.capacity),
             reverse=True)
         if not possible_datastores:
-            raise Exception("No possible datastores!")
+            raise Exception("No datastores are not available for provision, check status of "
+                            "datastore(s)")
         return possible_datastores[0]
 
+    def _pick_datastore_cluster(self):
+        """Pick a datastore cluster based on free space."""
+        possible_datastore_clusters = [dsc for dsc in self.system.get_obj_list(vim.StoragePod)
+                                       if dsc.overallStatus != "red"]
+        possible_datastore_clusters.sort(
+            key=lambda dsc: float(dsc.summary.freeSpace) / float(dsc.summary.capacity),
+            reverse=True)
+        if not possible_datastore_clusters:
+            raise Exception("No datastore clusters are not available for provision, check status "
```

Comment:
I think it would be better to raise `wrapanapi.exceptions.ItemNotFound` here, instead of the generic Exception